### PR TITLE
Habitevent add optional comment

### PR DESCRIPTION
**Pull request by j-vent**
Add error checking and handling for user to add an optional comment of 20 chars or less to a habit event.

## Description of Changes
Add error checking to activity
Intent test to check if validation works as intended

## Screenshots

error_comment

## Checklist

- [x] Automated tests
- [x] Screenshots included (if necessary)
- [x] Code is well commented

Closes #ISSUE_ID

### CODE REVIEW

**Comment by Docker-J** [comment]
How about set a maxlength of EditText in the layout? We can use android:maxLength attribute. IMO, it is better to keep the code as simple as possible

**Comment by j-vent** [comment]
Yeah i totally forgot that was a thing...Can I just keep it? It's already tested and everything :'(

**Review by loravocado — APPROVED** [review]

**Review by MacMalainey — DISMISSED** [review]
Addressed in a different PR I believe so I'll approve this one but HabitEventModel needs documentation

**Review by cheetiantey — DISMISSED** [review]
LGTM

File: src/app/src/main/java/com/cmput301f21t09/budgetprojectname/HabitEventModel.java [hunk 1]
```diff
@@ -12,22 +12,22 @@
 public class HabitEventModel{
     private final String ID; // TODO: change type depending on Firebase ID type
     private final String location; // TODO: change to Location once this is implemented
-    private final String description;
+    private final String comment;
     private final Date date;
     private final Image image;
 
-    HabitEventModel(String location, Date date, String description, Image image) {
+    HabitEventModel(String location, Date date, String comment, Image image) {
```
Comment (loravocado) [inline src/app/src/main/java/com/cmput301f21t09/budgetprojectname/HabitEventModel.java:19]: Please add Javadoc, but it looks excellent otherwise :)❤️

Comment (j-vent) [reply src/app/src/main/java/com/cmput301f21t09/budgetprojectname/HabitEventModel.java:19]: Added it in another PR!

**Review by loravocado — DISMISSED** [review]

**Review by MacMalainey — DISMISSED** [review]

**Review by MacMalainey — APPROVED** [review]

**Review by cheetiantey — APPROVED** [review]

**Review by AtrazineC — APPROVED** [review]

**Pull request merged**